staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -855,11 +900,23 @@ func decode(codec runtime.Codec, versioner storage.Versioner, value []byte, objP
 	if err := versioner.UpdateObject(objPtr, uint64(rev)); err != nil {
 		klog.Errorf("failed to update object version: %v", err)
 	}
+	if clusterName != "" {
```

Comment:
Same, anything this deep in the tree (where it shouldn't be) should be guarded by the HACK and talk about intent.